romancal/skymatch/skymatch_step.py

```diff
@@ -0,0 +1,217 @@
+                # NOTE: In principle we could assume that level is 0 and
+                # possibly add a log entry documenting this, however,
+                # at this moment I think it is safer to quit and...
+                #
+                # report inconsistency:
+                if self._is_asn:
+                    image_model.close()
+
+                raise ValueError(
+                    "Background level was subtracted but the "
+                    "'level' property is undefined (None)."
+                )
+
+            if image_model.meta["background"]["subtracted"] and self.subtract:
+                # cannot run 'skymatch' step on already "skymatched" images
+                # when 'subtract' spec is inconsistent with
+                # meta.background.subtracted:
+                if self._is_asn:
+                    image_model.close()
+
+                raise ValueError(
+                    "'subtract' step's specification is "
+                    "inconsistent with background info already "
+                    "present in image '{:s}' meta.".format(image_model.meta.filename)
+                )
+
+        wcs = deepcopy(image_model.meta.wcs)
+
+        sky_im = SkyImage(
+            image=image_model.data,
+            wcs_fwd=wcs.forward_transform,
+            wcs_inv=wcs.backward_transform,
+            pix_area=1.0,  # TODO: pixel area
+            convf=1.0,  # TODO: conv. factor to brightness
+            mask=dqmask,
+            id=image_model.meta.filename,  # file name?
+            skystat=self._skystat,
+            stepsize=self.stepsize,
+            reduce_memory_usage=self._is_asn,
+            meta={"image_model": input_image_model},
+        )
+
+        if self._is_asn:
+            image_model.close()
+
+        if self.subtract:
+            sky_im.sky = level
+
+        return sky_im
+
+    def _set_sky_background(self, sky_image, step_status):
+        image = sky_image.meta["image_model"]
+        sky = sky_image.sky
+
+        if self._is_asn:
+            dm = rdd.open(image)
+        else:
+            dm = image
+
+        if step_status == "COMPLETE":
+            # TODO: remove this block when ``rad`` has a background schema:
+            # https://github.com/spacetelescope/rad/issues/247
+            # This is a temporary workaround to access a 'background'
+            # entry into metadata as a Python dict, which we'll later define with
+            # a schema in ``rad``:
+            dm.meta["background"]["method"] = str(self.skymethod)
+            dm.meta["background"]["level"] = sky
+            dm.meta["background"]["subtracted"] = (
+                self.subtract or dm.meta["background"]["subtracted"]
+            )
+
+            if self.subtract:
+                dm.data[...] = sky_image.image[...]
+
+        dm.meta.cal_step.skymatch = step_status
+
+        if self._is_asn:
+            dm.save(image)
+            dm.close()
```

Comment:
Are you saving the infrastructure to add this to the pipeline for another ticket? (Possibly for a new pipeline - ELPP vs HLPP.)